release-team/hack/sync-enhancements-github-project-beta.sh

```diff
@@ -0,0 +1,164 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Usage: `sync-enhancements-github-project-beta.sh`
+
+set -e
+set -u
+set -o pipefail
+
+
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+REPO_ROOT="$(cd "${SCRIPT_DIR}/.." && pwd )"
+TMPDIR="${TMPDIR:-${REPO_ROOT}/tmp}"
+
+MILESTONE_ISSUES_JSON="${TMPDIR}/milestone_issues.json"
+mkdir -p "$(dirname "${MILESTONE_ISSUES_JSON}")"
+
+
+ORGANIZATION='kubernetes'
+REPOSITORY='enhancements'
+PROJECT_NUMBER=${GITHUB_PROJECT_BETA_NUMBER}
+MILESTONE=${RELEASE_MILESTONE}
+
+milestone_issue_ids=()
+milestone_issue_numbers=()
+
+
+# function definitions
+function get_field_ids_from_github_beta_project() {
+    if [ $1 == 'project-id' ]
+    then
+        query='.data.organization.projectV2.id'
+    else
+        ##for 'issue-number-field-id'
+        query='.data.organization.projectV2.fields.nodes[] | select(.name== "Issue Number") | .id'
+    fi
+    
+    ID="$( gh api graphql -f query='
+                    query($org: String!, $number: Int!) {
+                    organization(login: $org){
+                        projectV2(number: $number) {
+                        id
+                        fields(first:100) {
+                            nodes {
+                            ... on ProjectV2Field {
+                                id
+                                name
+                            }
+                            ... on ProjectV2SingleSelectField {
+                                id
+                                name
+                                options {
+                                    id
+                                    name
+                                }
+                            }
+                            }
+                        }
+                        }
+                    }
+                    }' -f org=${ORGANIZATION} -F number=${PROJECT_NUMBER} --jq "$query")"
+    echo $ID
+}
+
+echo 'Startig sync...'
+echo -e "[INFO] Fetching the list of open KEP issues from k/enhancements under the current release milestone, ${MILESTONE}"
+
+
+## Fetch k/enhancements issues & process json data
+gh api graphql -f query='
+            query($org: String!, $repo: String!, $milestone: String!){
+              repository(owner: $org, name: $repo) {
+                description
+                url
+                milestones(states: [OPEN],first:1, query: $milestone) {
+                  nodes{
+                    issues(states:[OPEN], first: 200){
```

Comment:
It's been a while since I've worked with graphql ... we are limiting the result set to 200 and also returning hasNextPage - are we able to check hasNextPage to see if there are more results (over 200)?  More of a nit since I don't think we've ever seen more than 100 KEPs opt in.

Reply:
you need to pass in `--paginage` to the `gh api graphql` call for the endCursor/hasNextPage values to be meaningful.

I commented in the bug-triage script w/ more details
https://github.com/kubernetes/sig-release/pull/1969/files#r948197296